src/BackGeoOracle.sol

```diff
@@ -161,8 +169,17 @@ contract BackGeoOracle is BaseHook {
             // the following condition must always be true since we only support ExactInput swaps
             assert(backAmountUnspecified < 0 && backAmountSpecified > 0);
 
+            address backrunAmountRecipient;
+
+            // retrieve original sender. The calling router is expected to implement msgSender() method
+            try IMsgSender(sender).msgSender() returns (address originalSender) {
```

Comment:
Still reviewing this with Vinicius to see what his thoughts are. Mine personally are that I like this solution, it's elegant, puts some of the responsibility on the user only to use a verified router, and removes that centralization vector of adding an owner. He also had some thoughts, but I asked him to add them directly, so he'll add them here when he can. As for me, this is a good fix.